docs: improve grammar, clarity, and code examples and API docs

dev-kas opened the pull request:
This PR updates the README and related docs to fix grammar issues, clarify language, and correct minor code example errors. The changes improves readability and consistency without changing any functionality.

Improvements:
- Fixed punctuation and grammar across multiple sections
- Rewrote informal or unclear phrases for increased clarity
- Corrected code examples (e.g., missing `fmt.Printf` format strings)
- Improved consistency in function descriptions

These updates make the docs more professional and accessible for both new and experienced.

dev-kas committed:
docs: improve grammar, clarity, and code examples and API docs

- fixed grammar and punctuation issues
- improved clarity and consistency in phrases
- corrected minor code example errors

stevenh commented on README.md:
```diff
@@ -29,7 +29,7 @@ Get a value out of the VM
 if value, err := vm.Get("abc"); err == nil {
     if value_int, err := value.ToInteger(); err == nil {
-        fmt.Printf("", value_int, err)
+        fmt.Println(value_int)
```
bug: you have lost the err.

stevenh commented on README.md:
```diff
@@ -170,7 +170,7 @@ The following are some limitations with otto:
 * `use strict` will parse, but does nothing.
 * The regular expression engine ([re2/regexp](https://pkg.go.dev/regexp)) is not fully compatible with the ECMA5 specification.
-* Otto targets ES5. Some ES6 features e.g. Typed Arrays are not supported, PR's to add functionality are always welcome.
+* Otto targets ES5. Some ES6 features, e.g., Typed Arrays, are not supported. Pull requests to add functionality are always welcome.
```
suggestion: no change needed. Omitting the comma is fine, only considered more correct in **American** English

stevenh commented on README.md:
````diff
@@ -283,15 +283,15 @@ var ErrVersion = errors.New("version mismatch")
 ```
 
-An Error represents a runtime error, e.g. a `TypeError`, a `ReferenceError`, etc.
+An Error represents a runtime error, e.g., a `TypeError`, a `ReferenceError`, etc.
````
suggestion: no change needed. Omitting the comma is fine, only considered more correct in **American** English

stevenh commented on README.md:
```diff
@@ -410,8 +410,8 @@ func (self Object) Set(name string, value interface{}) error
 
-An error will result if the setting the property triggers an exception (i.e.
-read-only), or there is an error during conversion of the given value.
+An error will result if setting the property triggers an exception (e.g.,
```
suggestion: no need for comma after e.g.

stevenh reviewed (changes requested):
Thanks for the PR, no need for commas after e.g. when using real english 😃

dev-kas committed:
docs: fix comma usage after 'e.g.' in documentation

- Remove commas after 'e.g.' per maintainer feedback
- Improve error handling in example code
- Maintain consistency in documentation style

Addresses feedback from PR #548

dev-kas commented:
Thanks for the feedback! Ive removed the commas after "e.g." as suggested. I also enhanced the error handling in the first code example for better clarity.

Let me know if there's anything else you'd like me to adjust.

stevenh commented on README.md:
```diff
@@ -745,8 +748,8 @@ func (self Value) Export() (interface{}, error)
-Export returns an error, but it will always be nil. It is present for backwards
-compatibility.
+Export returns an error, which will always be nil. It is included for backwards
+compatibility
```
nit: lost the full stop.

stevenh reviewed (changes requested):
Just a missing full stop and then good to go, thanks for the quick turn around.

dev-kas committed:
docs: add missing period in Export function documentation

Adds missing period at the end of the sentence in the Export function's
documentation to maintain consistent punctuation style.

dev-kas commented:
I've added the missing period to the Export function's documentation. The Pr should now be ready for your review.

stevenh commented:
Thanks for the PR @dev-kas
